ex4/ex4.c

```diff
@@ -7,9 +7,20 @@
 #include <unistd.h>
 #include <stdlib.h>
 
-int main(int argc, char* argv[])
+int main(int argc, char *argv[])
 {
-    // Your code here    
+    printf("hello world (pid: %d)\n", (int) getpid());
+    
+    int rc = fork();
+    if (rc < 0) {   
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+    } else if (rc == 0) {    
+        printf("hello, child here (pid: %d) \n", (int) getpid());
+    } else {
+        int ret;
```

Comment:
Not much point in initializing a variable to store the result of your `exec` call, since a successful `exec` doesn't return anything. This is because the process that calls `exec` gets transformed into the program that you specified in your call to `exec`, and that program has its own return values.